### ISSUE TITLE
Support AWS Glue Schema Registry for MSK

### ISSUE THREAD

**Issue by nmeagan11** [MaterializeInc/materialize#12304]
### Use Case

For users using AWS MSK, AWS Glue Schema Registry allows you to manage the schemas as resources in Terraform. When applying schema changes, the CI/CD process for Terraform will automatically apply the change to AWS, and all schemas are centralized and codified in source control.

If using the CSR, the primary interface for modifying schemas is a REST API. It becomes difficult to manage the schemas in source control and automatically integrate with the REST API if you self-host CSR.

**Comment by heeringa** [comment]
https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html

**Comment by morsapaes** [comment]
Related: flagging that, once we switch to Platform, the `USING SCHEMA FILE` option will no longer be around and there will be no way for users to provide a schema for Protobuf messages unless there's integration with Glue. Not sure how common this setup is + I'm aware that Kinesis isn't a priority for sources, but wanted to flag in any case because it might imply retracting Protobuf support for that particular case.

**Comment by heeringa** [comment]
Could users place a schema file on S3 or a public URL and point to that instead?  Could we provide support in an account for folks to upload or copy paste schemas and then make them available via some slick UX?

**Comment by morsapaes** [comment]
Both options would require engineering time and changes that might not be useful outside of the Kinesis source, so it doesn't sound like something worth getting into. At least not more worth than supporting Glue, which would at least also be useful for MSK. 

Do we know of _any_ users relying on schema files instead of a registry for non-Kinesis sources?

**Comment by heeringa** [comment]
> Do we know of any users relying on schema files instead of a registry for non-Kinesis sources?

Yeah, this is the right question to be asking.  None that I know of.  😄
